tools/target-server-validator/callout/src/main/java/PortOpenCheck.java

```diff
@@ -72,12 +95,34 @@ private static String available(final String host, final int port) {
   public ExecutionResult execute(final MessageContext messageContext,
     final ExecutionContext executionContext) {
     try {
-      String hostname = messageContext.getMessage().getHeader("host_name");
-      String port = messageContext.getMessage().getHeader("port_number");
-      int portnumber = Integer.parseInt(port);
-      String status = available(hostname, portnumber);
-      messageContext.setVariable("flow.reachableStatus", status);
-      return ExecutionResult.SUCCESS;
+      String payload = (String) messageContext.getVariable("request.content");
+      if (payload != null) {
+        Map<String, List<Map<String, String>>> result = new HashMap<>();
+        result.put("hostname_portnumbers_status", new ArrayList<>());
+        Map<String,List<Map<String,String>>> outerMap  = JavaxJson.fromJson(payload,Map.class);
```

Comment:
What's the motivation for implementing a generic json parser if we know the json format to be expected. Can't  we traversing the known structure with a JsonReader.readMap() and continue from there the same way as you do with the  explicit mapping operations below?

Reply:
+1 Can we not simply parse the entire JSON into HashMaps and ArrayLists with a single operation? I think this approach is suitable if unmarshalling into custom classes, but gson (and other libraries) should natively parse into HashMaps and ArrayLists.

Reply:
Right, I have updated the library to Gson and removed the generic parsing.